index.html

```diff
@@ -1866,12 +1851,15 @@ <h3> 'Subject Only' Terms of Use </h3>
       </p>
 
   <p>
-The Subject Only Terms of Use states that a verifiable credential MUST only be
-presented to a verifier by the subject. If a verifier is presented with a
-verifiable credential containing the Subject Only Terms of Use, by anyone other
-    than the subject, it MUST refuse to accept it.
+The Subject Only Terms of Use states that a verifiable credential is only to be presented by the subject. If a verifier is presented with a verifiable presentation containing a verifiable credential with a Subject Only Terms of Use, by anyone other
+    than the subject, it relies on that presentation on its own responsibility. Similarly, if a holder other than the subject presents such a presentation, they do so on their own responsibility.
      </p>
 
+	   <p class="note">
+There is an open question about whether or not the issuer can bind the verifier to an agreement embodied in a Terms of Use. This is especially problematic because the verifier is unable to view or commit to the terms before receiving the information covered by those terms. Necessarily, they receive the information before they can agree to the terms. In contrast, the holder is well aware of the Terms of Use prior to creating the presentation and therefore may be more likely to be held liable. However, this specification is not legal advice. Anyone relying on the legal foundation of a Terms of Use policy should seek the advice of competent counsel in the relevant jurisdictions.
+      </p>
```

Comment:
Needs CR/LFs adding otherwise it is unreadable.

As a general point, verification software can be configured to know before hand which ToUs it supports and which it does not support. So it is perfectly feasible for a verifier's implementation to either accept or reject a VC after parsing it, checking its contents, and determining if it is acceptable or not. Your note does not adequately address the point that the verifier needs to read a VC and its ToU before deciding whether to accept them or not. Your note implies that the mere act of reading and parsing a VC, and then rejecting it, is somehow committing to the ToUs.

Reply:
Thanks for the point about linefeeds. I'll make an update that corrects for that.

Yes. You could whitelist ToUs. But that isn't in the specification and I doubt we'd have consensus on requiring that method.

But I think you missed my point because whitelisting doesn't address the ordering of agreement (or not) to the ToU after receiving the information.

Any information given to me before I agree to a ToU will NOT be covered by that ToU if I choose to reject the ToU. What you're arguing for is a contract of adhesion https://legaldictionary.net/adhesion-contract/, aka, a shrink-wrap license https://www.wilmerhale.com/en/insights/publications/the-origin-of-click-wrap-software-shrink-wrap-agreements-march-22-2000. Historically, the boundary on the bindability of shrink-wrap licenses required an explicit additional step after the presentation of the license before the reveal of the licensed information. VCs don't have that. They especially don't have any way to record acceptance or not.

Those articles highlight the problems both for courts and for organizations dealing with shrink-wrap licenses. To wit:

> It is important to note, however, that there will be situations in which Internet companies may be well-advised not to use click-wrap agreements. For example, click-wrap agreements are more suitable when end-users are expected to be individuals, rather than organizations. Potential problems arise when a company, rather than an individual, enters into a click-wrap agreement. In this situation, the online merchant must be careful to ensure that the individual clicking to accept has power and authority to accept on behalf of the company. 

Your proposed requirement would suggest that the standard itself binds the recipient of a VC to certain actions (non-acceptance of the VC) in the case they don't understand or don't like a ToU. In effect, your language would create a potential legal liability for recipients of a VC should they reject the ToU. 

But... what constitutes acceptance and non-acceptance? Does forwarding the VC to legal counsel for review of the ToU constitute acceptance? What about ingesting it into a database, flagged for review but not used in active services? Does not acceptance require immediate purging? What about log files used for security and capacity analysis? Is it in appropriate to retain a VC to try to understand why it triggered non-acceptance?

What are the business and legal tests for compliance with any given ToU? Is the simple phrase "Subject must equal Holder" supposed to encapsulate a formal legal framework recognizable by a court of law? What is the burden of proof that the Subject *is* the Holder? And who has it? How would a verifier meet that standard? Is username/password acceptable or does it require the mythical DID-Auth? If the DID keys are stolen, does that leave the verifier not liable for the breach? Can the verifier simply rely on the good faith assertion of the holder? The verifier doesn't have a legal relationship with the issuer, so why are they on the hook when a holder impersonates someone else?

What are the remedies for violations and who has grounds for action? Is it only the issuer who can seek damages? Or can the verifier seek damages from the holder for misrepresentation? Can the issuer actually go after a verifier--with whom they have no relationship--or is their only remedy to sue the holder masquerading as the subject? What about the subject? Can they go after the issuer or the holder when a VC ToU is violated? Should the issuer have required a more rigorous proof? Did the verifier fail to meet the requirements and caused harm to the subject? What about the state? Does the state have grounds to pursue "improper acceptance" of a VC? Under what laws? False advertising, like current ToUs? Or privacy? This gets jurisdictional immediately. Is the verifier bound by the definitions of "accept" in the country in which they operate? Or does the location of the issuer, subject, or holder also create jurisdictional exposure?

In short, this is a minefield of legal liability. One which, IMO, makes such a requirement a complete non-starter.

Like a terms of use on a website, the ToU in a VC is a communication of policy. That's all. What recipients do in response to that policy is their business. It is NOT the business of the software to enforce.

Even courts acknowledge that the initial load of a web page doesn't constitute agreement to the sites ToU--because it is not possible to know what ToU, if any, is applicable before viewing that page. Continued use of a website, AFTER the opportunity to review the ToU can be interpreted as a binding click-wrap license... but VCs don't even have that. 

Could you imagine if web browsers refused to render web pages that had Terms of Use the user didn't understand or want? It would break the web. 

We simply can't put these kinds of requirements in the spec.

Reply:
The whole language of "accept" or "reject" is irrelevant.  The VC/VP is either verifiable or not.  In this case, the Presentation wouldn't verify with a code of "NotPresentedBySubject".  The Verifier has discretion over disposal of the payload. Section 8.9 Fitness for Purpose describes the obligation, discretion, and risks for the verifier in an appropriately "hand wavy" way. What the Verifier does with the presented artifact following verification isn't material to the Data model.

For ToU that are machine readable and can be part of the verification scheme, verification success/failure is appropriate.  For ToU that are are machine readable and aren't purely verification, then we're in the realm of protocol and judgement, so the DataModel can't express an opinion.  For ToU that aren't machine readable, the DM only needs to provide a method for transmission and the market will have to sort it out...

Reply:
@stonematt It sounds like you want acceptance of a ToU to be a part of verification:
> For ToU that are machine readable and can be part of the verification scheme, verification success/failure is appropriate.

My contention is that the specification doesn't provide any means of assurance about adherence to a ToU. We also  don't have anything in the data model that enables recording acceptance of a ToU (a Verifiable Presentation Receipt for example) . That might be a way to bootstrap a data-model centric way of testable commitment to a ToU, but it hasn't been a part of our conversation to date and certainly would be out of scope for 1.0.0. (A future specification for claims requests, credential reciepts, and presentation receipts seems like a good set of future work.)

If verification requires acceptance of the ToU, then we are making the verification process dependent on the issuer and/or holder, depending on the ToU, without a data model that has any way to represent acceptance must less the parties to the ToU. In contrast, if verification is purely about the cryptographic proof and procedural check (for revocation/status), then VCs can be verified independent of the issuer, holder, or verifier, as long as the key resolution and status resolution systems are running.

If verification requires agreement to a ToU, then the ToU needs to specify, and verification needs to programmatically prove, the counter parties to the ToU and their acceptance. In the Subject Must Be Holder ToU, are those terms binding on the Holder or the Verifier? How do we know if either party accepts the ToU? All the legal questions I raised above now become part of verification, turning it into, IMO, an intractable, unsolved problem.

As long as acceptance is not formally part of verification, then ToUs at either stage can state anything... which is fair. There will be use cases we won't imagine in our wildest inspirations and flexible ToUs are going to have to be part of that. 

But if acceptance is part of verification, then I ask, humbly, which fields in the data model are used to check if the correct counterparty (or parties) has accepted the ToU?

Reply:
Acceptance is not formally part of verification.  My primary point was to say the language of "accept" and "reject" are out of bounds.  That concept requires the judgement of the parties involved.  Acceptance is a Fit-for-Purpose decision that requires more context that the VP alone.  A Verifier may reject a VP b/c they don't trust the issuer, or the credential isn't appropriate.  In either cases, the VP verified.  

ToU are policy statements that stakeholders must follow or bear the risk of not following (see verification step 8.9). Some may be expressed in a machine readable way, others won't be. Our narrow scope is to provide something verifiable. At this stage of the process, perhaps no ToU can be "verifiable" in an automated way because we haven't adopted such a technology, and are thus advisory.  In the event the market adopts such a technology, parsing a ToU will result in one of 3 outcomes. 

1. ToU may explicitly pass verification (verifier can read it, reason over it, and determine whether it's fit for purpose and choose to comply), 
2. explicitly fail verification (one of the above is explicitly false), 
3. result in an "unknown" status  (based on ToU content,  details included in the VP and perhaps some future protocol) 

The verifier ultimately decides what to do with the artifact that was presented, regardless of the Acceptance decision. Also, I agree that the compelling idea of receipts (of any kind) definitely don't fit in v1.0 of this spec.

Can you suggest a path forward (quickly)? Otherwise, this will be published as an at-risk feature and likely fall out of the spec before publication.

Reply:
I'll update ASAP based on the feedback so far. 

Sadly, I'm still a bit confused about your position. It seems like you're trying to leave the door open for language that verification depends on the ToU, but that you aren't recommending any spec-text that says that at this time.

If you look at Section 8.9 in the PR, I replaced "MUST be enforced" with "SHOULD be respected" to separate conformant, testable verification (which I see as the "MUST" requirements) from business rules, policies, and their enforcement. 

If you're ok with that language, I think we're good.